packages/core/src/index.ts

```diff
@@ -255,7 +292,15 @@ export function createEnv<
         ..._shared,
       };
 
-  const parsed = parseWithDictionary(finalSchema, runtimeEnv);
+  const parsed = opts.createFinalSchema
+    ? opts
+        .createFinalSchema(finalSchemaShape as never, isServer)
+        ["~standard"].validate(runtimeEnv)
+    : parseWithDictionary(finalSchemaShape, runtimeEnv);
+
+  if (parsed instanceof Promise) {
+    throw new Error("Validation must be synchronous");
+  }
```

Comment:
this is double - maybe `parseWithDictoionary` should just accept the `createFinalSchema` function and handle the transformation internally to keep this DRY?

Reply:
the check in parseWithDictionary is per key, so it would still need to be separate

could have an `ensureSynchronous(result)` helper though?

Reply:
pushed an `ensureSynchronous` helper up - doesn't save *much* code but it's a little more semantic to look at